lib/cog/adapters/http/adapter_bridge.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Cog.Adapters.Http.AdapterBridge do
+  @moduledoc """
+  Mediates interactions between HTTP requests and pipeline executions
+  """
+
+  use GenServer
+  use Adz
+
+  def start_link,
+    do: GenServer.start_link(__MODULE__, [], name: __MODULE__)
+
+  def submit_request(cog_user_name, id, initial_context, pipeline, timeout) do
+    try do
+      GenServer.call(__MODULE__,
+                     {:submit_request, cog_user_name, id, initial_context, pipeline},
+                     timeout)
+    catch
+      :exit, {:timeout,_} ->
+        {:error, :timeout}
+    end
+  end
+
+  def finish_request(room, response),
+    do: GenServer.call(__MODULE__, {:finish_request, room, response})
+
```

Comment:
I think a timeout of `:infinity` might be better on both `GenServer.call`s. Since there's only one of these bridge processes, and it's one of the few Cog components exposed to potential high load (assuming most triggers are activated via m2m traffic), there's the possibility these calls could timeout which could lead to some interesting failure scenarios.